File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> mp = new HashMap<>();

        for(int num : nums){
            mp.put(num, mp.getOrDefault(num, 0)+1);
        }

        int i = 0;
        int[] ans = new int[k];
        while(k > 0){
            int max = -1;
            int num = -1;

            for(int key : mp.keySet()){
                int freq = mp.get(key);

                if(freq > max){
                    max = freq;
                    num = key;
                }
            }

            ans[i] = num;
            mp.remove(num);
            i++;
            k--;
        }

        return ans;
    }
}

File: Data Structures & Algorithms/house-robber/submission-3.py
class Solution:
    def rec(self, i, nums, dp):
        if i >= len(nums):
            return 0
        if dp[i] != -1:
            return dp[i]

        take = nums[i] + self.rec(i+2, nums, dp)
        
        not_take = self.rec(i+1, nums, dp)

        dp[i] = max(take, not_take)
        return dp[i]

    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [-1]*n
        return self.rec(0, nums, dp)